src/openms/include/OpenMS/ANALYSIS/ID/FalseDiscoveryRate.h

```diff
@@ -114,6 +158,488 @@ namespace OpenMS
+    {
+      if (id.getIdentifier() == identifier)
+      {
+        setScores_(scores_to_FDR, id, score_type, higher_better, keep_decoy, charge);
+      }
+    }
+
+    template <typename IDType>
+    void setScores_(const std::map<double,double>& scores_to_FDR, IDType & id, const std::string& score_type,
+                    bool higher_better, bool keep_decoy, const String& identifier) const
+    {
+      if (id.getIdentifier() == identifier)
+      {
+        setScores_(scores_to_FDR, id, score_type, higher_better, keep_decoy);
+      }
+    }
+
+    void setScores_(const std::map<double,double>& scores_to_FDR, PeptideIdentification & id, const std::string& score_type,
+                    bool higher_better, int charge, const String& identifier) const
+    {
+      if (id.getIdentifier() == identifier)
+      {
+        setScores_(scores_to_FDR, id, score_type, higher_better, charge);
+      }
+    }
+
+    template <typename IDType>
+    void setScores_(const std::map<double,double>& scores_to_FDR, IDType & id, const std::string& score_type,
+                    bool higher_better, const String& identifier) const
+    {
+      if (id.getIdentifier() == identifier)
+      {
+        setScores_(scores_to_FDR, id, score_type, higher_better);
+      }
+    }
+
+
+    //TODO could also get a keep_decoy flag when we define what a "decoy group" is -> keep all always for now
+    void setScores_(
+        const std::map<double,double>& scores_to_FDR,
+        std::vector<ProteinIdentification::ProteinGroup>& grps,
+        const std::string& score_type,
+        bool higher_better) const;
+
+    static void checkTDAnnotation_(const MetaInfoInterface & idOrHit)
+    {
+        if (!idOrHit.metaValueExists("target_decoy"))
+        {
+          throw Exception::MissingInformation(__FILE__,
+                                              __LINE__,
+                                              OPENMS_PRETTY_FUNCTION,
+                                              "Meta value 'target_decoy' does not exist in all ProteinHits! Reindex the idXML file with 'PeptideIndexer'");
+        }
+    }
+
+    template <typename HitType>
+    struct GetLabelFunctor: std::function<bool(const HitType&)>
+    {
+      bool operator() (const HitType& hit)
+      {
+        //TODO if we checked in the beginning, this check could be skipped.
+        if (!hit.metaValueExists("target_decoy"))
+        {
+          throw Exception::MissingInformation(__FILE__,
+                                              __LINE__,
+                                              OPENMS_PRETTY_FUNCTION,
+                                              "Meta value 'target_decoy' does not exist in all ProteinHits! Reindex the idXML file with 'PeptideIndexer'");
+        }
+        else
+        {
+          return std::string(hit.getMetaValue("target_decoy"))[0] == 't';
+        }
+      }
+    };
+
+    template <typename HitType>
+    struct TrueFunctor: std::function<bool(const HitType&)>
```

Comment:
what is this good for? also below the false functor?

Reply:
to disable / enable filterings?

Reply:
For separate TD DBs. Removed, since support with the new structure was not complete.